src/nexus-sdk-base.ts

```diff
@@ -0,0 +1,63 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+import { NexusSDKConfig } from './types';
+
+/**
+ * Base class for the Nexus SDK.
+ * Handles configuration and provides common functionality for all SDK components.
+ */
+export class NexusSDKBase {
+  protected apiUrl: string;
+
+  /**
+   * Create a new instance of NexusSDKBase
+   * @param config SDK configuration
+   */
+  constructor(config: NexusSDKConfig = {}) {
+    this.apiUrl = config.apiUrl ?? 'https://api.nexusmutual.io/v2';
+  }
+
+  /**
+   * Sends an HTTP request to the specified endpoint
+   * @param endpoint API endpoint to send the request to
+   * @param options Axios request configuration
+   * @returns Promise resolving to the response data
+   */
+  // TODO: validation if options is empty? empty options might lead to axios failing
```

Comment:
should we drop this comment? looks like the code has a fallback to handle all other cases